tests/conftest.py

```diff
@@ -969,7 +969,8 @@ def pytest_runtest_makereport(item, call):
     if call.when == 'setup':
         item.user_properties.append(('start', str(datetime.fromtimestamp(call.start))))
     elif call.when == 'teardown':
-        log_custom_msg(item)
+        if item.nodeid == item.session.items[-1].nodeid:
```

Comment:
shouldn't we move the logic to sessionfinish hook?

Reply:
Tried couple of methods, which didn't work, but I think I have another idea I want to test it out. Let me get back to this

Reply:
Based on my research, it seems impossible to update user_properties or results.xml in pytest_sessionfinish. "Appending values to item.user_properties during pytest_sessionfinish won't work because the item objects have already been processed and their results are being finalized."

Other option is to manually get result.xml values and append what we want, then write the file from pytest_sessionfinish, which is not the best idea imo